contributors/design-proposals/scheduling/scheduling-policy.md

```diff
@@ -0,0 +1,302 @@
+
+Also Mandatory values must also be enforced by scheduling policies in case of:
+
+-  mutli-arch (amd64, arm64) of multi-os (Linux, Windows) clusters (will also be handled later by the [RuntimeClass]() KEP)
+-  multi-az / region / failure domain clusters
+
+# Overview
+
+### syntaxic standards
+
+### empty match and unset fields:
+
+We follow Kubernetes conventions, an empty field matches everything an unset one matches nothing
+
+### how policies are computed
+
+Policies are computed using the following algorithm:
+
+```
+sortedPolicies = sort_by_priority(sort_deny_first(policies))
+for policy in sortedPolicies:
+  if policy matches pod: // all specified policy rules match
+    return policy.action
+```
+ note that rules of policies from a lower priority are superseeded by ones from a higher priority if they match.
+
+
+
+# Detailed Design
+
+
+## SchedulingPolicy
+
+Proposed API group: `policy/v1alpha1`
+
+
+### SchedulingPolicy content
+
+SchedulingPolicy spec is composed of optional fields that allow scheduling rules. If a field is absent from a SchedulingPolicy, this `SchedulingPolicy` won't allow any item from the missing field.
+
+```yaml
+apiVersion: policy/v1alpha1
+kind: SchedulingPolicy
+metadata:
+  name: mySchedulingPolicy
+spec:
+  priority: <exception,cluster,default>
+  action:  <allow,deny>     
+  rules:
+    namespaceSelector:
+      - key: value
+        operator: operator
+    podSelector:
+        key: value
+    schedulerNames:
+       match: []
+    priorityClassNames:
+       match: []
+    tolerations:
+       match: []
+    nodeSelectors:
+       match: []
+    nodeAffinities:
+       match: []
+    podAntiAffinities:
+       match: []
+    podAffinities:
+       match: []
+```
+
+- allow: When a policy matches the pod is allowed
+- deny: when a policy matches the pod is denied
+
+
+### Scoping
+
+The `spec.rules.namespaceSelector` and `spec.rules.podSelector` attributes scope the policy application range.
+
+Pods must match both `spec.namespaceSelector` and `spec.podSelector` matching rules to be constrained by a policy.
+Both `spec.namespaceSelector` and `spec.podSelector` are optional. If absent, all pods and all namespaces are targeted by the policy.
+
+### Scheduler name
+
+If `schedulerNames` is absent from `allowed`, `default` or `required`, no scheduler is allowed by this specific policy.
+
+
+#### Allowed
+Allow pods to use either the `green-scheduler` (which is used by specifying `spec.schedulerName` in pod definition) or the `my-scheduler` scheduler (by specifying `spec.schedulerName: "my-scheduler"`) in namespaces labeled `team`:
+
+```yaml
+apiVersion: policy/v1alpha1
+kind: SchedulingPolicy
+metadata:
+  name: mySchedulingPolicy
+spec:
+  action: allow    
+  rules:
+    namespaces:
+      - key: team
+        operator: Exists
+    podSelector: {}
+    schedulerNames:
+       match: ["green-scheduler","my-scheduler"]
+```
+
+An empty list of schedulerNames will allow usage of all schedulers:
```

Comment:
Remove this. It's redundant with what's already been stated.

Reply:
True.